Tests/NIOHTTP1Tests/HTTPDecoderLengthTest.swift

```diff
@@ -0,0 +1,519 @@
+            XCTFail("Unexpected error: \(error)")
+        }
+
+        // Must spin the loop.
+        (channel.eventLoop as! EmbeddedEventLoop).run()
+    }
+
+    private func assertRequestWithInvalidCLErrors(contentLengthField: String) throws {
+        XCTAssertNoThrow(try channel.pipeline.add(handler: HTTPRequestDecoder()).wait())
+
+        // Send a GET with the invalid headers.
+        do {
+            try channel.writeInbound(ByteBuffer(string: "POST / HTTP/1.1\r\nContent-Length: \(contentLengthField)\r\n\r\n"))
+            XCTFail("Did not throw")
+        } catch HTTPParserError.invalidContentLength {
+            // ok
+        } catch HTTPParserError.unexpectedContentLength {
+            // also ok
+        } catch {
+            XCTFail("Unexpected error: \(error)")
+        }
+
+        // Must spin the loop.
+        (channel.eventLoop as! EmbeddedEventLoop).run()
+    }
+
+    func testRequestWithMultipleDifferentContentLengthsFails() throws {
+        try assertRequestWithInvalidCLErrors(contentLengthField: "4, 5")
+    }
+
+    func testRequestWithMultipleDifferentContentLengthsOnDifferentLinesFails() throws {
+        try assertRequestWithInvalidCLErrors(contentLengthField: "4\r\nContent-Length: 5")
+    }
+
+    func testRequestWithInvalidContentLengthFails() throws {
+        try assertRequestWithInvalidCLErrors(contentLengthField: "pie")
+    }
+
+    func testRequestWithIdenticalContentLengthRepeatedErrors() throws {
+        // This is another case where http_parser is, if not wrong, then aggressively interpreting
+        // the spec. Regardless, we match it.
+        XCTAssertNoThrow(try channel.pipeline.add(handler: HTTPRequestDecoder()).wait())
+
+        // Send two POSTs with repeated content length, one with one field and one with two.
+        // Both should error.
+        do {
+            try channel.writeInbound(ByteBuffer(string: "POST / HTTP/1.1\r\nContent-Length: 4, 4\r\n\r\n"))
+            XCTFail("Did not throw")
+        } catch HTTPParserError.invalidContentLength {
+            // ok
+        } catch {
+            XCTFail("Unexpected error \(error)")
+        }
+
+        // Must spin the loop.
+        (channel.eventLoop as! EmbeddedEventLoop).run()
+    }
+
+    func testRequestWithMultipleIdenticalContentLengthFieldsErrors() throws {
+        // This is another case where http_parser is, if not wrong, then aggressively interpreting
+        // the spec. Regardless, we match it.
+        XCTAssertNoThrow(try channel.pipeline.add(handler: HTTPRequestDecoder()).wait())
+
+        do {
+            try channel.writeInbound(ByteBuffer(string: "POST / HTTP/1.1\r\nContent-Length: 4\r\nContent-Length: 4\r\n\r\n"))
+            XCTFail("Did not throw")
+        } catch HTTPParserError.unexpectedContentLength {
+            // ok
+        } catch {
+            XCTFail("Unexpected error \(error)")
+        }
+
+        // Must spin the loop.
+        (channel.eventLoop as! EmbeddedEventLoop).run()
```

Comment:
assert that the channel is not active anymore and also call `channel.finish()` and assert return value.

Reply:
Nope, we can't do that here because `channel.finish()` will throw if the channel is already closed.